src/HPolygon.jl

```diff
@@ -55,7 +62,14 @@ HPolygon{N}() where {N<:Real} = HPolygon{N}(Vector{LinearConstraint{N}}())
 HPolygon() = HPolygon{Float64}()
 
 # conversion constructor
-HPolygon(S::LazySet) = convert(HPolygon, S)
+function HPolygon(S::LazySet; validate_boundedness::Bool=false)
```

Comment:
this function is strange. i think that we should eliminate it.

Reply:
The reason we have them is probably this warning in v0.7:
```julia
julia> H = Hyperrectangle(low=N[-1, -1], high=N[1, 1]);

julia> HPolygon(H)
┌ Warning: Constructors no longer fall back to `convert`.
A constructor `HPolygon(::Hyperrectangle{Float64})` should be defined instead.
```
If you still find it necessary, let us do this change somewhere else.

Reply:
these constructors are old. i wrote that we should remove them in the past but i don't find the issue.

Reply:
We lose a feature if we remove them. Why do we want that?

Reply:
`convert(HPolygon, S)` should be used in favor of `HPolygon(S)` IMO. we have these `convert` aliases only for the `HPolygon`s.